packages/gatsby-plugin-sharp/src/process-file.js

```diff
@@ -229,8 +229,11 @@ const compressJpg = (pipeline, outputPath, options) =>
   )
 
 const compressWebP = (pipeline, outputPath, options) =>
-  pipeline.toBuffer().then(sharpBuffer =>
-    imagemin
+  pipeline.toBuffer().then(sharpBuffer => {
+    if (process.platform === `win32`) {
+      return fs.writeFile(outputPath, sharpBuffer)
+    }
+    return imagemin
       .buffer(sharpBuffer, {
         plugins: [
```

Comment:
What's going on here? Why have you modified this function to write the buffer to disk on Windows but continue the problem on other platforms?

Don't modify this method, it does not need to be run, the file will still be written to disk. Instead the condition earlier that calls the method should only call it if the user has configured it to do so(either globally in `gatsby-config` or via a new graphql param `compress`/`minify` and corresponding format only variants like `quality` does. Assuming @wardpeet agrees with that approach first.